### PR TITLE
Move custom domain away from app object

### PR DESCRIPTION
**Pull request by masvelio**
## Link to Issue
Closes: #8299 

## Description of Changes
- removed all custom-domain-related attributes from app object 
- moved the logic to react-query
- simplified useInitApp hook by moving fetching domain query to initApp function

## Test Plan
- to test custom domain locally you need to follow the instructions from [here](https://github.com/hicommonwealth/commonwealth/blob/624f91366e7f9f827bafbf6335d44a8b8642fa50/common_knowledge/Custom-Domains.md#local-testing) 
- go to custom domain eg `https://forums.dydx.community` and click around, everything should work 

## Deployment Plan

1. n/a

## Other Considerations

- n/a

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/finish_social_login.tsx [hunk 1]
```diff
@@ -10,7 +11,9 @@ import { PageLoading } from 'views/pages/loading';
 const validate = async (
   setRoute: (route: string) => void,
   isLoggedIn: boolean,
+  isCustomDomain?: boolean,
 ) => {
+  console.log('isCustomDomain', isCustomDomain);
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/finish_social_login.tsx:16]: lets remove

File: packages/commonwealth/client/scripts/views/pages/finish_social_login.tsx [hunk 2]
```diff
@@ -58,23 +61,28 @@ const FinishSocialLogin = () => {
   const navigate = useCommonNavigate();
   const [validationError, setValidationError] = useState<string>('');
   const user = useUserStore();
+  const { data: domain } = useFetchCustomDomainQuery();
+
+  console.log('dziala?');
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/finish_social_login.tsx:66]: console

**Review by mzparacha — APPROVED** [review]

File: packages/commonwealth/client/scripts/views/pages/finish_social_login.tsx (hunk 1 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/finish_social_login.tsx]: 🙈

File: packages/commonwealth/client/scripts/views/pages/finish_social_login.tsx (hunk 2 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/finish_social_login.tsx]: 🤦

**Review by Israellund — APPROVED** [review]
LGTM

**Pull request merged**